app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesRepositoryFirestore.kt

```diff
@@ -0,0 +1,202 @@
+package com.github.se.icebreakrr.model.profile
+
+import android.util.Log
+import com.google.android.gms.tasks.Task
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.GeoPoint
+
+class ProfilesRepositoryFirestore(private val db: FirebaseFirestore) : ProfilesRepository {
+
+  private val collectionPath = "profiles"
+
+  /**
+   * Generates a new unique profile ID from Firestore.
+   *
+   * @return A new unique profile ID as a String.
+   */
+  override fun getNewProfileId(): String {
+    return db.collection(collectionPath).document().id
+  }
+
+  /**
+   * Initializes the repository by setting an authentication state listener.
+   *
+   * @param onSuccess A callback that is invoked when the user is authenticated.
+   */
+  override fun init(onSuccess: () -> Unit) {
+    Firebase.auth.addAuthStateListener {
+      if (it.currentUser != null) {
+        onSuccess()
+      }
+    }
+  }
+
+  /**
+   * Temporary implementation: Retrieves all profiles in the Firestore database.
+   *
+   * @param center The center point for the radius query (currently unused).
+   * @param radiusInMeters The radius around the center (currently unused).
+   * @param onSuccess A callback invoked with a list of profiles if the operation is successful.
+   * @param onFailure A callback invoked with an exception if the operation fails.
+   */
+  override fun getProfilesInRadius(
+      center: GeoPoint,
+      radiusInMeters: Double,
+      onSuccess: (List<Profile>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    // TODO Add GeoFirestore functionality
+    val task = db.collection(collectionPath).get()
+
+    task.addOnCompleteListener { result ->
+      if (result.isSuccessful) {
+        val profiles =
+            result.result?.documents?.mapNotNull { document -> documentToProfile(document) }
+                ?: emptyList()
+
+        onSuccess(profiles)
+      } else {
+        result.exception?.let { e ->
+          Log.e("ProfilesRepositoryFirestore", "Error getting profiles", e)
+          onFailure(e)
+        }
+      }
+    }
+  }
```

Comment:
This allows to get all users and not just the ones in radius.
But as we mentioned during meetings, for testing purposes, we'll maintain it like this for S1 and implement geofirestore radius sorting later.